### PR TITLE
Lots of styling and git improvements

### PR DESCRIPTION
**Pull request by iHiD**
I would **not** merge this yet, as it's not tested with the development-environment setup and will therefore almost certainly break everything for y'all. I'll check it tomorrow before I merge.

---

Further styling to exercise page and tabs.

Screenshot 2020-10-26 at 22 40 08

---

This PR also has a commit carving out the git stuff.

### CODE REVIEW

File: app/commands/parse_markdown.rb [hunk 1]
```diff
@@ -22,25 +25,25 @@ def call
...
+      %i[table tagfilter strikethrough]
+    )
+    renderer.render(html)
   end
 
+  memoize
   def preprocessed_text
-    @preprocessed_text ||=
-      text.gsub(/^`{3,}(.*?)`{3,}\s*$/m) { "\n#{$&}\n" }
+    text.gsub(/^`{3,}(.*?)`{3,}\s*$/m) { "\n#{$&}\n" }
```
Comment (ErikSchierboom) [inline app/commands/parse_markdown.rb:46]: This does something with lines starting with a backtick? Regexes can be awful to read :D

File: app/commands/submission/create.rb [hunk 2]
```diff
@@ -6,12 +6,12 @@ def initialize(solution, files, submitted_via)
       @submission_uuid = SecureRandom.compact_uuid
 
       @solution = solution
-      @files = files
+      @submitted_files = files
```
Comment (ErikSchierboom) [inline app/commands/submission/create.rb:9]: Nice naming improvement!

File: app/helpers/view_components/track/generic_nav.rb [hunk 3]
```diff
@@ -7,14 +7,14 @@ def to_s
...
-                Exercism::Routes.tracks_path,
-                class: "back",
-                'aria-label': "Back to all tracks"
-              ),
+              # link_to(
+              #   graphical_icon("arrow-left"),
+              #   Exercism::Routes.tracks_path,
+              #   class: "back",
+              #   'aria-label': "Back to all tracks"
+              # ),
```
Comment (ErikSchierboom) [inline app/helpers/view_components/track/generic_nav.rb:19]: Does this need a TODO indicating why the code is commented?

File: app/models/git/exercise.rb [hunk 4]
```diff
@@ -1,64 +1,66 @@
...
         solution.git_sha
       )
     end
 
-    def initialize(repo, slug, commit, track_config)
-      @repo = repo
-      @slug = slug
-      @commit = commit
-      @track_config = track_config
+    def initialize(track_slug, exercise_slug, git_sha)
```
Comment (ErikSchierboom) [inline app/models/git/exercise.rb]: Is this a candidate for Mandate?

File: app/models/git/exercise.rb [hunk 5]
```diff
@@ -1,64 +1,66 @@
...
+      @track_slug = track_slug
+      @exercise_slug = exercise_slug
+      @git_sha = git_sha
     end
 
-    def filepaths
-      files.map { |defn| defn[:full] }
+    def data
```
Comment (ErikSchierboom) [inline app/models/git/exercise.rb:17]: You'll likely have already considered this, but is it worth memoizing this? I don't know if there are places where callers will call this method more than once. If not, ignore this.

File: test/serializers/serialize_solution_test.rb [hunk 6]
```diff
@@ -29,39 +32,43 @@ class SerializeSolutionTest < ActiveSupport::TestCase
...
-    refute valid_filepath =~ track.repo.ignore_regexp
-    assert ignore_filepath =~ track.repo.ignore_regexp
-
-    output = SerializeSolution.(solution, solution.user)
-    assert_includes output[:solution][:files], valid_filepath
-    assert_includes output[:solution][:files], ignore_filepath
-  end
+  # TODO: I think we don't need this test any more as it was
+  # only checking the exercise files, which are now checked elsewhere.
+  # But I'm leaving it here for now in case I was wrong.
```
Comment (ErikSchierboom) [inline test/serializers/serialize_solution_test.rb:37]: :)

**Review by ErikSchierboom — APPROVED** [review]

File: app/commands/parse_markdown.rb (hunk 1 above)
Comment (iHiD) [reply app/commands/parse_markdown.rb:46]: Ha. Yes. I've not actually written this here - just changed the memoization. This whole file will probably get written if we switch to highlight.js. I have no idea what the regexp does either 😬

File: app/helpers/view_components/track/generic_nav.rb (hunk 3 above)
Comment (iHiD) [reply app/helpers/view_components/track/generic_nav.rb:19]: Done :)

File: app/models/git/exercise.rb (hunk 4 above)
Comment (iHiD) [reply app/models/git/exercise.rb]: Done. Thanks!

File: app/models/git/exercise.rb (hunk 5 above)
Comment (iHiD) [reply app/models/git/exercise.rb:17]: Nice. yes. Done :)

**Comment by iHiD** [comment]
Merging this. @ErikSchierboom @kntsoriano This uses the new git server which requires the development-environment to be updated as per https://github.com/exercism/development-environment/pull/41

**Pull request merged**